fvm/evm/types/offchain.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import "github.com/onflow/flow-go/model/flow"
+
+// BackendStorageSnapshot provides a read only view of registers
+type BackendStorageSnapshot interface {
+	GetValue(owner []byte, key []byte) ([]byte, error)
+}
+
+// StorageProvider provides access to storage at
+// specific time point in history of the EVM chain
+type StorageProvider interface {
+	// GetSnapshotAt returns a readonly snapshot of storage
+	// at specific block (start state of the block before executing transactions)
+	GetSnapshotAt(height uint64) (BackendStorageSnapshot, error)
```

Comment:
since this is an EVM component inside FVM can we name the height either `flowHeight` or `evmHeight` otherwise its sometimes not obvious. We might want to introduce types as well.

Reply:
good idea, going to rename to evmBlockHeight

Reply:
renamed